src/main/java/servlets/module/lesson/XxeLesson.java

```diff
@@ -0,0 +1,174 @@
+package servlets.module.lesson;
+
+import dbProcs.FileInputProperties;
+import dbProcs.Getter;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+import org.owasp.encoder.Encode;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import utils.ShepherdLogManager;
+import utils.Validate;
+import utils.XmlDocumentBuilder;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.*;
+import javax.xml.parsers.DocumentBuilder;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+/**
+ * Cross Site Scripting Lesson
+ * <br/><br/>
+ * This file is part of the Security Shepherd Project.
+ * <p>
+ * The Security Shepherd project is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.<br/>
+ * <p>
+ * The Security Shepherd project is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.<br/>
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with the Security Shepherd project.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author ismisepaul
+ */
+public class XxeLesson
+        extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+    private static Logger log = Logger.getLogger(XxeLesson.class);
+    private static final String LEVEL_NAME = "XXE Lesson";
+    private static final String LEVEL_ID = "df2ac757cc135dcb8ce5ea01f677c74f04b446d6";
```

Comment:
Should this be pulled out of the DB via using the Hash? I don't think Level ID's are hard coded anywhere else off the top of my head

Reply:
Switched to grabbing it from the level hash (yeah it's not being done anywhere else)

https://github.com/OWASP/SecurityShepherd/blob/2cac9ef1673aa8716253a645be66ef3db4145a58/src/main/java/servlets/module/lesson/XxeLesson.java#L58